contracts/pcv/PCVDepositAggregator.sol

```diff
@@ -0,0 +1,365 @@
+
+                // Only withdraw from this underlying if it has an overage 
+                if (actualPcvDepositBalance > idealPcvDepositBalance) {
+                    uint pcvDepositOverage = actualPcvDepositBalance - idealPcvDepositBalance;
+                    uint amountToWithdraw = pcvDepositOverage;
+                    
+                    if (totalAmountNeeded < pcvDepositOverage) {
+                        amountToWithdraw = totalAmountNeeded;
+                    }
+                    
+                    IPCVDeposit(pcvDepositAddress).withdraw(address(this), amountToWithdraw);
+                    totalAmountNeeded -= amountToWithdraw;
+
+                    // If we don't need to withdraw anymore, stop looping over the deposits
+                    if (totalAmountNeeded == 0) break;
+                } else {
+                    continue;
+                }
+            }
+
+            IERC20(token).safeTransfer(to, amount);
+        }
+    }
+
+    function withdrawERC20(address _token, address to, uint256 amount) external virtual override onlyPCVController {
+        IERC20(_token).safeTransfer(to, amount);
+    }
+
+    function withdrawETH(address payable to, uint256 amount) external virtual override onlyPCVController {
+        to.transfer(amount);
+    }
+
+    function setBufferWeight(uint128 newBufferWeight) external virtual override onlyGuardianOrGovernor {
+        int128 difference = int128(newBufferWeight) - int128(bufferWeight);
+        bufferWeight = uint128(int128(bufferWeight) + difference);
+
+        totalWeight = uint128(int128(totalWeight) + difference);
+    }
+
+    function setPCVDepositWeight(address depositAddress, uint128 newDepositWeight) external virtual override onlyGuardianOrGovernor {
+        if (!pcvDepositAddresses.contains(depositAddress)) {
+            revert("Deposit does not exist.");
+        }
+
+        int128 difference = int128(newDepositWeight) - int128(pcvDepositInfos[depositAddress].weight);
+        pcvDepositInfos[depositAddress].weight = uint128(newDepositWeight);
+
+        totalWeight = uint128(int128(totalWeight) + difference);
+    }
+
+    function removePCVDeposit(address pcvDeposit) external virtual override onlyGuardianOrGovernor {
+        _removePCVDeposit(address(pcvDeposit));
+    }
+
+    function addPCVDeposit(address newPCVDeposit, uint256 weight) external virtual override onlyGovernor {
+        _addPCVDeposit(address(newPCVDeposit), uint128(weight));
+    }
+
+    function setNewAggregator(address newAggregator) external virtual override onlyGovernor {
+        // Add each pcvDeposit to the new aggregator
+        for (uint i=0; i < pcvDepositAddresses.length(); i++) {
+            address pcvDepositAddress = pcvDepositAddresses.at(i);
+            uint128 pcvDepositWeight = pcvDepositInfos[pcvDepositAddress].weight;
+
+            IPCVDepositAggregator(newAggregator).addPCVDeposit(pcvDepositAddress, pcvDepositWeight);
+        }
```

Comment:
If the other PCVDeposit aggregator reverts, should check existence before adding